### ISSUE TITLE
Option twoside

### ISSUE THREAD

**Issue by r4sky0** [Digital-Media/HagenbergThesis#1]
Hallo,

kurze Frage: Wenn ich meine Arbeit binden lassen möchte, empfiehlt sich gleich das ganze beidseitig zu bedrucken. Ist in dem Paket der beidseitige Druck (twoside) vorgesehen, damit jeweils der linke und rechte Rand abwechselnd einen größeren Abstand zur Bindung bekommen? 

Ich habe bereits gesehen, dass das Paket auf der Klasse "book" aufsetzt, allerdings passen beim setzen der twoside Option z. B. die Seitenüberschriften nicht mehr. Gibt's da eine Lösung?

**Comment by imagingbook** [comment]
Nein, derzeit nicht, in der Vorlage ist nur einseitiger Druck vorgesehen.
--WB

**Comment by TimDiercks** [comment]
Gibt es hierzu inwischen eine Lösung?

**Comment by imagingbook** [comment]
Here is a quick patch to switch to `twopage` format in the current `hgbthesis` setup:

1) File `hgbthesis.cls`:
	Change 
		`\LoadClass[a4paper,11pt,oneside]{book}`
	to
		`\LoadClass[a4paper,11pt,twoside]{book}`
		
2) File `hgbheadings.sty`:
	Change
		`\fancyhead[R]{\sffamily\thepage}`
		`\fancyhead[L]{\sffamily\nouppercase{\leftmark}}`
	to
		`\fancyhead[RO,LE]{\sffamily\thepage}` 
		`\fancyhead[LO]{\sffamily\nouppercase{\rightmark}}`
		`\fancyhead[RE]{\sffamily\nouppercase{\leftmark}}`

This will show 

- 	on EVEN (left-side) pages: the current **page number** + **chapter title**,
- 	on ODD (right-side) pages: the current **section title** + **page number**.

Header contents can be customized by modifying `\leftmark` and `\rightmark` (e.g., using the `\markboth` command). This opens many options as of what people would like to see in the two headers. For example, some may want to see the author's name and thesis title there instead. So while the technical path is straightforward, a more general solution would add quite some overhead. I therefore suggest to keep the one-page setup at this time.

**Comment by hochleitner** [comment]
I'll post this here as evidence of how other universities are handling this topic. University of Vienna obviously requires two-sided prints when handing in a thesis: https://mtbl.univie.ac.at/storage/media/mtbl02/2023_2024/2023_2024_62.pdf

On page 2 it says:

> (2) Die Arbeit ist im DIN A4-Hochformat, hart gebunden und doppelseitig bedruckt einzureichen.

Maybe we should consider passing the parameter to the class options to enable this at some point.

**Comment by imagingbook** [comment]
I agree, but as said its not only about passing the option through. I'm happy to look at this once the current changes are in done ...

**Comment by imagingbook** [comment]
The recent commit (c3c3fc8d1d6274fd5cc6f6a270665c2ac1e3995f) to branch `two-page-option` adds simple option handling to `hgbthesis.cls`. Final header/footer setup (in `hgbheadings.sty`) still needs to be done.

**Comment by imagingbook** [comment]
All relevant modifications for handling `oneside` and `twoside` options have been migrated to branch `flexible-cover-page` (commit d31e158696e3486debecbfca9cb9d79a0fb90bdd) for easier merging back into `main`. Branch `two-page-option` is thus obsolete and will be removed.
All `hagenberg-thesis` classes now accept and handle these options, `oneside` still being the default, as before.

**Comment by hochleitner** [comment]
Aw, this is beautiful. ❤️
Let's see how much chaos it creates, when people do two-sided documents when one sided is recommended.
Maybe we should add something to the tutorial? E.g. "For digital-only documents use one-sided. Use two-sided for printed theses when required by your program or university."

**Comment by imagingbook** [comment]
I wondered about this too. 
How about adding a check to the theme to issue a warning if the wrong option is used?

**Comment by hochleitner** [comment]
Theme checks could be an idea, but would need a clear commitment on how to handle it. I'm not sure if can can manage to get this even for our university. 😬

**Comment by imagingbook** [comment]
Well, the chaos would be the same if people used vanilla LaTeX ...
